### PR TITLE
docs: & style: Update readme requirements

### LINKED ISSUE #92: Use requirements.txt instead of requirements-dev
To align with the contents of the readme and python standard practices the requirements.txt file should not be empty. It should match the dev in the main branch. If dev is different it should be in a feature branch to test dependency upgrades.

### LINKED ISSUE #93: Add pip tools to the readme
To use the pip-compile dependency management flow we need to add the necessity for pip tools into the readme.md file

### PR DESCRIPTION
**Pull request by hlipsig**
## This PR
Updates the readme to include the pip-tools dependency and a link to it and corrects an install command that was incorrect. 

- adds the dependency list to requirements.in and also has the compiled version in requirements.txt

### Related Issues

Fixes #92 
Fixes #93 

### Notes
Readme still needs to be updated to explain how to use the requirements-dev file vs requirements file and the pip-compile command and dependency workflow.

### Follow-up Tasks
Create an issue for the task in Notes.
address issue #94 

### How to test
Steps in readme will now succeed.

### CODE REVIEW

File: requirements.in [hunk 1]
```diff
@@ -0,0 +1,9 @@
+pylint
+pytest
+pytest-mock
+black
+pip-tools
+pre-commit
+flake8
+pytest-mock
+coverage
```
Comment (tcarrio) [inline requirements.in:9]:
```suggestion
```

We won't want to add these development dependencies into our main `requirements.txt`. These are already defined in our `requirements-dev.txt`, which is relevant to contributors specifically.

File: requirements.txt [hunk 2]
```diff
@@ -1,6 +1,103 @@
 #
-# This file is autogenerated by pip-compile with python 3.10
```
Comment (tcarrio) [inline requirements.txt:2]: We can probably revert the changes to the `requirements.txt` file as well, since it's an artifact of the `requirements.in` file.

However! This would be a good time to check if items like the `setuptools` pinning will happen with the `requirements.in` file being empty :grin:

File: requirements.txt [hunk 3]
```diff
@@ -1,6 +1,103 @@
...
+    # via pre-commit
+wheel==0.38.4
+    # via pip-tools
+wrapt==1.14.1
+    # via astroid
+
+# The following packages are considered to be unsafe in a requirements file:
+# pip
+# setuptools
+setuptools>=65.5.1 # not directly required, pinned by Snyk to avoid a vulnerability
```
Comment (tcarrio) [inline requirements.txt:103]: This one specifically is interesting, so I think we should generate the `requirements.txt` still even with the empty `requirements.in` as I mentioned.

**Comment by tcarrio** [comment]
Hey @hlipsig! Thank you for opening this PR. Great catch on the installation instructions, the `-r` flag is absolutely necessary :sweat_smile: 

Please see my code review comments regarding the `requirements.in` file and the compiled `requirements.txt`, as we'll want to make the necessary changes there to ensure we aren't enforcing development dependencies outside of our development processes.

**Comment by tcarrio** [comment]
@hlipsig also dove into the the issues you had opened and have some further feedback on #92 regarding the `requirements.txt` and `requirements-dev.txt` approach.

**Comment by tcarrio** [comment]
From the `pip-tools` documentation, it would seem we can eliminate our `requirements.in` and `requirements-dev.in` entirely and use our `pyproject.toml` as the source of truth for the `requirements*.txt` artifacts. I included more information in that issue. Let me know what you think, and thank you again for your contribution! :smile:

**Comment by beeme1mr** [comment]
Thanks for the pr @hlipsig and for your detailed feedback @tcarrio. It sounds like we may be able to just use pyproject.toml and add most of the dependencies in the dev group.

https://python-poetry.org/docs/master/managing-dependencies/#adding-a-dependency-to-a-group

Is that what you had in mind @tcarrio?

**Comment by hlipsig** [comment]
> Thanks for the pr @hlipsig and for your detailed feedback @tcarrio. It sounds like we may be able to just use pyproject.toml and add most of the dependencies in the dev group.
> 
> https://python-poetry.org/docs/master/managing-dependencies/#adding-a-dependency-to-a-group
> 
> Is that what you had in mind @tcarrio?

If we're open to moving to .toml then that's great and ideal. It felt like too big a change to suggest previously, but I should have.

File: requirements.txt (hunk 3 above)
Comment (hlipsig) [reply requirements.txt:103]: The Snyk checks failed without this. I do agree we should have minimum and maximum package version numbers to protect against vulnerabilities and guarantee success. That is a good best practice.

**Comment by beeme1mr** [comment]
This SDK is still sub 1.0, so I think it's fine to make these changes if we think this will be better in the long run.

**Comment by hlipsig** [comment]
I'm going to close this PR and then make some proposed changes based on what we've discussed here and on the issues. :)
